File: src/types/knowledge.ts
/** ナレッジエントリの注入モード */
export type InjectionMode = 'system_prompt' | 'tool_reference';

/** ナレッジエントリのメタデータ（content除外の軽量表現） */
export interface KnowledgeEntryMeta {
  id: string;
  file_name: string;
  size_bytes: number;
  enabled: boolean;
  injection_mode: InjectionMode;
  created_at: string;
}

File: src-tauri/src/models/knowledge.rs
use serde::{Deserialize, Serialize};

/// DB上のKnowledge_Entry完全表現
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KnowledgeEntry {
    pub id: String,
    pub session_id: String,
    pub file_name: String,
    pub content: String,
    pub size_bytes: i64,
    pub enabled: bool,
    pub injection_mode: String,
    pub created_at: String,
}

/// フロントエンド向け軽量表現（content除外）
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct KnowledgeEntryMeta {
    pub id: String,
    pub file_name: String,
    pub size_bytes: i64,
    pub enabled: bool,
    pub injection_mode: String,
    pub created_at: String,
}
